dev-docs/RFCs/v4.0.0/names_for_concepts_types_functions.md

```diff
@@ -0,0 +1,260 @@
+In the codebase and in the documentation, strictly correct terminology should *always* be used, as many functions and algorithms distinguish between hexagons and pentagons.
+
+
+## `H3_EXPORT` and C name collisions
+
+To avoid C name collisions, we should build the bindings (`h3-py`, `h3-java`, `h3-go`, and `h3-node`) with a standard function prefix, using the `H3_EXPORT` C macro.
+The `h3-js` binding would not need this.
+
+The proposed prefix is `h3_`.
+
+## Functions
+
+### Conventions for "property getters" and transforms
+
+- use `get` prefix for
+    + constant data (`getNumCells`, `getRes0Cells`)
+    + object properties (`getResolution`, `getBaseCellNumber`)
+- use `to` to denote transforms
+    + different representations of the same object
+    + when doing a lossy transformation to a new object (`cellToParent`, `pointToCell`)
+- do not use `get` or `to` for *computations*
+    + e.g., `polyfill`, `compact`, `cellAreaKm2`
+
+There is some ambiguity between property, transform, and computation, so use your best judgement with these guidelines in mind.
+
+### Validity checks
+
+- `isValid*` should mean that a *full* validity check is made
+- without `Valid` (like in the case of `isPentagon`), we do not guarantee
+  that a full validity check is made; instead, a user should assume only a
+  minimal bit mask check is done
+    - we could imagine a `isValidPengaton` function, if full verification
+      would be convenient
+- similarly, a function like `areNeighborCells` will assume each cell
+  has passed the `isValidCell` check; the function will do only minimal
+  work to determine if they are neighbors
+
+
+### General Function Names
+
+|          Current name         |     Proposed name      |
+|-------------------------------|------------------------|
+| *Does Not Exist (DNE)*        | `isValidIndex`         |
+| `h3IsValid`                   | `isValidCell`          |
+| `h3UnidirectionalEdgeIsValid` | `isValidDirectedEdge`  |
+| `h3IsPentagon`                | `isPentagon`           |
+| `h3IsResClassIII`             | `isResClassIII`        |
+| `h3IndexesAreNeighbors`       | `areNeighborCells`     |
+| `h3ToParent`                  | `cellToParent`         |
+| `h3ToChildren`                | `cellToChildren`       |
+| `numHexagons`                 | `getNumCells`          |
+| `getRes0Indexes`              | `getRes0Cells`         |
+| `getPentagonIndexes`          | `getPentagons`         |
+| `h3GetBaseCell`               | `getBaseCellNumber`    |
+| `h3GetResolution`             | `getResolution`        |
+| `geoToH3`                     | `pointToCell`          |
+| `h3ToGeo`                     | `cellToPoint`          |
+| `compact`                     | same, `compactCells`   |
+| `uncompact`                   | same, `uncompactCells` |
+| *DNE*                         | `getMode`              |
+| `polyfill`                    | same, `polygonToCells` |
```

Comment:
We need to come to a final conclusion before merging this RFC on these three. Do we rename them with the `Cells` nomenclature, or leave them as is?

Reply:
Resolved in the meeting today. Thanks!